Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/person/core/PersonCertificatesStorageTest.kt

```diff
@@ -1,69 +1,89 @@
-import androidx.core.content.edit
 import de.rki.coronawarnapp.covidcertificate.common.certificate.CertificatePersonIdentifier
+import de.rki.coronawarnapp.covidcertificate.person.core.PersonCertificatesSettings.Companion.CURRENT_PERSON_KEY
+import de.rki.coronawarnapp.covidcertificate.person.core.PersonCertificatesSettings.Companion.PERSONS_SETTINGS_MAP
 import de.rki.coronawarnapp.util.serialization.SerializationModule
 import io.kotest.matchers.shouldBe
 import io.mockk.MockKAnnotations
-import io.mockk.every
 import io.mockk.impl.annotations.MockK
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.test.TestCoroutineScope
+import kotlinx.coroutines.test.runBlockingTest
+import org.joda.time.Instant
 import org.junit.jupiter.api.BeforeEach
 import org.junit.jupiter.api.Test
 import testhelpers.BaseTest
+import testhelpers.TestDispatcherProvider
 import testhelpers.extensions.toComparableJsonPretty
-import testhelpers.preferences.MockSharedPreferences
+import testhelpers.preferences.FakeDataStore
 
 @Suppress("MaxLineLength")
 class PersonCertificatesStorageTest : BaseTest() {
     @MockK lateinit var context: Context
-    private lateinit var mockPreferences: MockSharedPreferences
+    private lateinit var fakeDataStore: FakeDataStore
+    private val personIdentifier1 = CertificatePersonIdentifier(
+        dateOfBirthFormatted = "01.10.2020",
+        firstNameStandardized = "fN",
+        lastNameStandardized = "lN"
+    )
+
+    private val personIdentifier2 = CertificatePersonIdentifier(
+        dateOfBirthFormatted = "20.10.2020",
+        firstNameStandardized = "ffNN",
+        lastNameStandardized = "llNN"
+    )
 
     @BeforeEach
     fun setup() {
         MockKAnnotations.init(this)
 
-        mockPreferences = MockSharedPreferences()
-
-        every {
-            context.getSharedPreferences("certificate_person_localdata", Context.MODE_PRIVATE)
-        } returns mockPreferences
+        fakeDataStore = FakeDataStore()
     }
 
     private fun createInstance() = PersonCertificatesSettings(
-        context = context,
-        gson = SerializationModule().baseGson()
+        dataStore = fakeDataStore,
+        mapper = SerializationModule.jacksonBaseMapper,
+        appScope = TestCoroutineScope(),
+        dispatcherProvider = TestDispatcherProvider()
     )
 
     @Test
-    fun `init is sideeffect free`() {
-        createInstance()
+    fun `init is sideeffect free`() = runBlockingTest {
+        createInstance().apply {
+            currentCwaUser.first() shouldBe null
+            personsSettings.first() shouldBe emptyMap()
+        }
     }
 
     @Test
     fun `clearing deletes all data`() {
-        mockPreferences.edit {
-            putString("deleteme", "test")
+        createInstance().apply {
+            setCurrentCwaUser(personIdentifier1)
+            setBoosterNotifiedAt(personIdentifier1)
+            setDccReissuanceNotifiedAt(personIdentifier1)
+            clear()
```

Comment:
Would be nice to additionally check before calling `clear()` that these fields are non-null: 

```kotlin
        fakeDataStore[CURRENT_PERSON_KEY] shouldNotBe null
        fakeDataStore[PERSONS_SETTINGS_MAP] shouldNotBe null
```

Reply:
done